playbooks/backup.yaml

```
@@ -0,0 +1,48 @@
+# Playbook: Set up backups.
+---
+- name: Set up backups
+  hosts: libvirt_hosts
+  become: true
+  tasks:
+    - name: Install needed packages
+      ansible.builtin.package:
+        name: restic
+        state: present
+
+    - name: Set up backup script
+      ansible.builtin.copy:
+        src: files/backup/backup.sh
+        dest: /usr/local/sbin/backup.sh
+        owner: root
+        group: root
+        mode: '0700'
+
+    - name: Retrieve restic password
+      delegate_to: localhost
+      ansible.builtin.command:
+        cmd: ../scripts/pass {{ inventory_hostname }}/restic
+        chdir: '{{ lookup("env", "PWD") }}'
+      become: false
+      register: backup__restic_pass
+      changed_when: false
+
+    - name: Save restic password
+      ansible.builtin.copy:
+        content: '{{ backup__restic_pass.stdout }}'
```

You don't need to execute pass manually, you can use the passwordstore lookup like this: `{{ lookup('community.general.passwordstore', inventory_hostname + '/restic') }}`. But also, both the details of where and how passwords are stored should be left for the caller to decide, in here you can assume you get the cleartext password in a variable

```suggestion
    - name: Save restic password
      ansible.builtin.copy:
        content: '{{ backup__restic_password }}'
```

---

Good to know about the `community.general.passwordstore`. Thanks!